org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/LambdaExpressionAndMethodRefFixCore.java

```diff
@@ -0,0 +1,490 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2022 Fabrice TIERCELIN and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Fabrice TIERCELIN - initial API and implementation
+ *******************************************************************************/
```

Comment:
It's always good to add a contributor note about how this code was refactored from xxxx to jdt.core.manipulation.  I usually change the copyright to only specify the current year since this is a new file but as you have noted, it must  retain the original copyright owner.  This patch otherwise looks good for 4.29 M1 when master is open.